packages/client/src/devices/MicrophoneManager.ts

```diff
@@ -61,6 +61,16 @@ export class MicrophoneManager extends InputMediaDeviceManager<MicrophoneManager
   private async startSpeakingWhileMutedDetection(deviceId?: string) {
     await this.stopSpeakingWhileMutedDetection();
     if (isReactNative()) {
+      // If there is no stream we create one and publish it first.
+      if (!this.call.publisher?.isPublishing(this.trackType)) {
+        const stream = await this.getStream({ deviceId });
```

Comment:
I think you should add a flag to `unmuteStream` to create a muted stream (the naming will be weird, that the `unmuteStream` function creates a muted stream, but I think we can live with that, or at least I don't have a better name)

And then call `unmuteStream` here, the reason:
- the `unmuteStream` also applies default constraints: https://github.com/GetStream/stream-video-js/blob/main/packages/client/src/devices/InputMediaDeviceManager.ts#L220
- register necessary event handlers, in case the stream end unexpectedly: https://github.com/GetStream/stream-video-js/blob/main/packages/client/src/devices/InputMediaDeviceManager.ts#L233

We will miss all those functionality if you create the stream like this

Reply:
maybe we can rename `unmuteStream` to `createOrUnmuteStream` and similarly `muteStream` could be `muteOrDestroyStream`

Reply:
One more thing: `this.call.publisher?.isPublishing(this.trackType)` -> only check if we're publishing, you should also check if `this.state.mediaStream === undefined`, and only create the stream if that's true as well